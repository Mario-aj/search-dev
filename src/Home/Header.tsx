import * as React from 'react';
import { useNavigate } from 'react-router-dom';

const Header = () => {
  const [isOpen, setIsOpen] = React.useState(false);
  const navigate = useNavigate();

  function handleLogout() {
    localStorage.removeItem('gh-dev-access_token');
    navigate('/login');
  }

  return (
    <div className="flex items-center justify-between w-full h-full max-w-4xl">
      <h1 className="text-2xl font-bold">Gh-dev</h1>
      <div className="relative" onClick={() => setIsOpen((open) => !open)}>
        <img
          src="https://scontent.flad1-1.fna.fbcdn.net/v/t39.30808-6/247089419_3097038823913123_4266727059060168780_n.jpg?_nc_cat=100&ccb=1-5&_nc_sid=09cbfe&_nc_ohc=oMpiSOjIhtMAX-fzHtS&_nc_ht=scontent.flad1-1.fna&oh=00_AT-7JhN3mhp7YMuNIH7VTSCE6I-BGuG1fqP0Bg4JxEa8vw&oe=61E95D24"
          alt="profile"
          className="w-12 h-12 rounded-full cursor-pointer"
        />

        <div
          title="dropdown"
          className={`absolute flex-col gap-2 p-3 px-2 text-gray-800 bg-white rounded-md top-14, ${
            isOpen ? 'flex' : 'hidden'
          }`}
        >
          <a href="#" className="px-2 cursor-pointer hover:bg-blue-200">
            Profile
          </a>
          <button
            type="button"
            onClick={handleLogout}
            className="bg-transparent cursor-pointer hover:bg-blue-200"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  );
};

export default Header;
