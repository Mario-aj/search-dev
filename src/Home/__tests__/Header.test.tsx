import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';

const MockHeader = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);

describe('Header', () => {
  describe('Behavior', () => {
    it('should render correctly', () => {
      render(<MockHeader />);

      const dropdown = screen.getByTitle('dropdown');
      const img = screen.getByRole('img', { name: /profile/i });

      expect(
        screen.getByRole('heading', { name: /Gh-dev/i })
      ).toBeInTheDocument();
      expect(img).toBeInTheDocument();
      expect(dropdown).toBeInTheDocument();
      expect(dropdown).toHaveClass('hidden');
    });

    it('should show dropdown and its children', () => {
      render(<MockHeader />);

      const dropdown = screen.getByTitle('dropdown');
      const img = screen.getByRole('img', { name: /profile/i });

      userEvent.click(img);
      expect(dropdown).not.toHaveClass('hidden');
      expect(dropdown).toHaveClass('flex');

      const link = screen.getByRole('link', { name: /profile/i });
      const button = screen.getByRole('button', { name: /logout/i });

      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', '#');
      expect(button).toBeInTheDocument();
      userEvent.click(button);
      expect(window.location.pathname).toBe('/login');
    });

    it('should call logout function', () => {
      render(<MockHeader />);

      const button = screen.getByRole('button', { name: /logout/i });

      userEvent.click(button);
      expect(window.location.pathname).toBe('/login');
    });
  });

  describe('Styles', () => {
    it('should render correctly', () => {
      render(<MockHeader />);

      const header = screen.getByRole('heading', { name: /Gh-dev/i });
      const img = screen.getByRole('img', { name: /profile/i });
      const dropdown = screen.getByTitle('dropdown');

      expect(header).toMatchSnapshot();
      expect(img).toMatchSnapshot();
      expect(dropdown).toMatchSnapshot();
      expect(dropdown).toMatchSnapshot();
    });
  });
});
