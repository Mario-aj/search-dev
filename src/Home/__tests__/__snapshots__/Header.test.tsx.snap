// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Styles should render correctly 1`] = `
<h1
  class="text-2xl font-bold"
>
  Gh-dev
</h1>
`;

exports[`Header Styles should render correctly 2`] = `
<img
  alt="profile"
  class="w-12 h-12 rounded-full cursor-pointer"
  src="https://scontent.flad1-1.fna.fbcdn.net/v/t39.30808-6/247089419_3097038823913123_4266727059060168780_n.jpg?_nc_cat=100&ccb=1-5&_nc_sid=09cbfe&_nc_ohc=oMpiSOjIhtMAX-fzHtS&_nc_ht=scontent.flad1-1.fna&oh=00_AT-7JhN3mhp7YMuNIH7VTSCE6I-BGuG1fqP0Bg4JxEa8vw&oe=61E95D24"
/>
`;

exports[`Header Styles should render correctly 3`] = `
<div
  class="absolute flex-col gap-2 p-3 px-2 text-gray-800 bg-white rounded-md top-14, hidden"
  title="dropdown"
>
  <a
    class="px-2 cursor-pointer hover:bg-blue-200"
    href="#"
  >
    Profile
  </a>
  <button
    class="bg-transparent cursor-pointer hover:bg-blue-200"
    type="button"
  >
    Logout
  </button>
</div>
`;

exports[`Header Styles should render correctly 4`] = `
<div
  class="absolute flex-col gap-2 p-3 px-2 text-gray-800 bg-white rounded-md top-14, hidden"
  title="dropdown"
>
  <a
    class="px-2 cursor-pointer hover:bg-blue-200"
    href="#"
  >
    Profile
  </a>
  <button
    class="bg-transparent cursor-pointer hover:bg-blue-200"
    type="button"
  >
    Logout
  </button>
</div>
`;
